lib/rules/no-constant-condition.js

```diff
@@ -45,11 +44,18 @@ module.exports = {
     },
 
     create(context) {
-        const options = context.options[0] || {},
-            checkLoops = options.checkLoops !== false,
-            loopSetStack = [];
+        const options = context.options[0] || {};
+        let checkLoops = options.checkLoops ?? "allExceptWhileTrue";
+        const loopSetStack = [];
         const sourceCode = context.sourceCode;
 
+        if (options.checkLoops === true) {
+            checkLoops = "all";
+        }
+        if (options.checkLoops === false) {
```

Comment:
Shall we use an `else if`? I think that makes the logic easier to follow, but it's just my feeling.

```suggestion
        } else if (options.checkLoops === false) {
```

Reply:
Done!